rs/sns/testing/src/sns.rs

```diff
@@ -438,12 +647,21 @@ pub mod pocket_ic {
         )
         .await;
 
-        super::upgrade_sns_controlled_test_canister(
+        super::propose_sns_controlled_test_canister_upgrade(
             &dev_participant_agent,
             sns,
             canister_id,
             upgrade_arg,
         )
-        .await;
+        .await
+    }
+    pub async fn wait_for_sns_controlled_canister_upgrade(
```

Comment:
Should this function be inlined?

Reply:
This function is used in PocketIC CI tests and in `run-basic-scenario`, so we still need an abstract `CallCanisters`-based implementation

Reply:
But we probably don't need PocketIC-specific version for it 😅